discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -451,102 +389,79 @@ impl SearchScope for SqliteStorage {
         search: &Search,
         documents: &[NewDocument],
     ) -> Result<(), Error> {
-        let mut tx = self.begin_tx().await?;
-        let mut query_builder = QueryBuilder::new("INSERT INTO ");
+        let mut tx = self.pool.begin().await?;
 
-        query_builder
-            .push(
-                "Search (rowid, searchBy, searchTerm, pageSize, pageNumber) VALUES (1, ?, ?, ?, ?)",
-            )
-            .build()
-            .persistent(false)
+        sqlx::query(
+            "INSERT INTO Search (rowid, searchBy, searchTerm, pageSize, pageNumber) VALUES (1, ?, ?, ?, ?)"
+        )
             .bind(search.search_by as u8)
             .bind(&search.search_term)
             .bind(search.paging.size)
             .bind(search.paging.next_page)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
+            .await?;
 
         if documents.is_empty() {
-            return Self::commit_tx(tx).await;
+            tx.commit().await?;
+            return Ok(());
         };
 
         SqliteStorage::store_new_documents(&mut tx, documents).await?;
-        query_builder
-            .reset()
-            .push("SearchDocument (documentId) ")
+
+        QueryBuilder::new("INSERT INTO SearchDocument (documentId) ")
```

Comment:
🔧 somewhat unrelated to the current change, i just noticed that this should be wrapped into a
```rust
for documents in documents.chunks(BIND_LIMIT) {}
```
loop. even though this case is quite unlikely, it's better to be safe. similarly in other methods where we can't guarantee that the `BIND_LIMIT` is not exceeded.

Reply:
IMHO I mentioned it before but I still think we should not bother with handling `BIND_LIMIT` it doesn't add any value for our use cases as in no situations will we have a SQL queries which inserts that many documents for the same user (like we currently add at most 10 in practice, and this is very unlikely to ever be over 100).  The bind limit does even by far out class the number of documents we ever want to have in the database at any point in time AFIK. 

This means that basically any situation where we have more then `BIND_LIMIT` documents (or other entities) is basically a unexpected bug-like situation where having an error should be ok.

The think is even if we do use chunking by `BIND_LIMIT`, it is very maintenance error prone as it is trivial to forget to update the divider used and in turn still run into an error because it runs out of possible binds.

This is also prone to cause a lot of maintenance/review friction over time (forgotten loop, forgotten updated divider). It also can any make any code using a `WHERE id not in <tuple>`  more complex for no practical benefit.

Either way I would say it is out of scope (:pushpin: ) of this PR as it would need to be changed in several places, not just here. And might entail further discussion.

Reply:
(as a side note this is one of the kind of situations I was asking about with the poll on slack, where you could fix it in this PR but I generally think its better to add a :pushpin: , maybe a task with maybe even a estimate and do it in a follow up)

Reply:
i'm ok either way, fully remove it or fully adapt it, but it should be handled uniformly across the code (in a different pr of course).